tests/cypress/tests/unit/lexeme_test.ts

```diff
@@ -0,0 +1,111 @@
+            expect(fieldExpression.resolve.bind(undefined, undefined, { value: '}' } as ISyntaxTree)).to.throw(Error);
+        }
+    });
+
+    it('resolves string expressions', () => {
+        expect(!!stringExpression.resolve).to.equal(true);
+        expect(typeof stringExpression.resolve).to.equal('function');
+
+        if (stringExpression.resolve) {
+            expect(stringExpression.resolve(undefined, { value: '\'abc\'' } as ISyntaxTree)).to.equal('abc');
+            expect(stringExpression.resolve(undefined, { value: '"abc"' } as ISyntaxTree)).to.equal('abc');
+            expect(stringExpression.resolve(undefined, { value: '`abc`' } as ISyntaxTree)).to.equal('abc');
+            expect(stringExpression.resolve(undefined, { value: '"\\""' } as ISyntaxTree)).to.equal('"');
+            expect(stringExpression.resolve(undefined, { value: `'\\''` } as ISyntaxTree)).to.equal(`'`);
+            expect(stringExpression.resolve(undefined, { value: '`\\``' } as ISyntaxTree)).to.equal('`');
+            expect(stringExpression.resolve(undefined, { value: '\'\\\'' } as ISyntaxTree)).to.equal('\\');
+
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '3' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: 'abc' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '{abc' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: 'abc}' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '}abc{' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '{{abc}}' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '"' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '`' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: `'` } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '{' } as ISyntaxTree)).to.throw(Error);
+            expect(stringExpression.resolve.bind(undefined, undefined, { value: '}' } as ISyntaxTree)).to.throw(Error);
+        }
+    });
+
+    it('resolves value expressions', () => {
+        expect(!!valueExpression.resolve).to.equal(true);
+        expect(typeof valueExpression.resolve).to.equal('function');
+
+        if (valueExpression.resolve) {
+            expect(valueExpression.resolve(undefined, { value: 'abc' } as ISyntaxTree)).to.equal('abc');
+            expect(valueExpression.resolve(undefined, { value: 'abc   ' } as ISyntaxTree)).to.equal('abc');
```

Comment:
do we drop leading spaces as well? what happens if the target data contains leading/trailing spaces? Is this behavior different for `stringExpression`? Also a test with spaces mid-string in `stringExpression` would be nice, to show that you (presumably) don't need to escape them.

Reply:
Adding escaped leading/trailing spaces cases. The underlying assumption for the lexemes is that the lexer will take care of trimming the string prior to trying to find a match.